### ISSUE TITLE
Recipe Robot doesn't work with special characters in app names

### ISSUE THREAD

**Issue by homebysix** [homebysix/recipe-robot#117]
Example provided by @jaharmi:

```
$ recipe-robot --verbose --ignore ~/Downloads/Touché.app 
    
Processing ~/Downloads/Touché.app ...
[ERROR] Recipe Robot exploded with unexpected error: 
Traceback (most recent call last):
  File "./scripts/recipe-robot", line 95, in main
    process_input_path(facts)
  File "./scripts/recipe_robot_lib/inspect.py", line 71, in process_input_path
    facts["input_path"] = input_path
  File "./scripts/recipe_robot_lib/facts.py", line 81, in __setitem__
    val = NotifyingString(self.default_suffix, val)
  File "./scripts/recipe_robot_lib/facts.py", line 143, in __new__
    instance = super(NotifyingString, cls).__new__(cls, text.encode("utf-8"))
UnicodeDecodeError: 'ascii' codec can't decode byte 0xcc in position 30: ordinal not in range(128)    
```

**Comment by homebysix** [comment]
Related: https://github.com/homebysix/recipe-robot/issues/197

**Comment by homebysix** [comment]
Resolved as of v2.3.0.
